src/modules/auth/passport/oidc-strategy.ts

```diff
@@ -39,16 +46,47 @@ passport.use(
       );
     }
 
+    if (
+        !process.env.OIDC_PROVIDER ||
+        !process.env.OIDC_CONFIG ||
+        !process.env.OIDC_CLIENT_ID ||
+        !process.env.OIDC_CLIENT_SECRET ||
+        !process.env.OIDC_REDIRECT_URI
+    ) {
+        logger.error('Not all OIDC environment variables are properly defined');
+        callback('Internal server error');
+    }
+
+    if (Object.values(OidcProviders).includes(process.env.OIDC_PROVIDER as OidcProviders)) {
+        logger.error('The environment variable OIDC_PROVIDER is not a valid OIDC provider. Options are ' + Object.values(OidcProviders).join('; '));
+        callback('Internal server error');
+    }
+
+    let oidcConfig: OidcConfig;
+
+    try {
+        oidcConfig = await new AuthService().getOIDCConfig();
+    } catch (e) {
+        logger.error(e);
+        callback('Internal server error');
+    }
+
+    if(!oidcConfig!) {
```

Comment:
If you return after all callbacks, this check if not needed.

```suggestion
    if(!oidcConfig) {
```